File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            
            // If the complement exists, we found our pair
            if (mp.containsKey(complement)) {
                return new int[]{mp.get(complement), i+1};
            }
            
            // Otherwise, put the current number and index in the map
            mp.put(nums[i], i+1);
        }
        
        return new int[]{0, 0}; 
    }
}

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
    mp := make(map[int]int)
    ans1:=0
    ans2:=0
    for i:=0;i<len(numbers);i++{
        if(mp[target - numbers[i]]!=0){
            ans1=mp[target - numbers[i]]
            ans2=i+1
        }
        mp[numbers[i]] = i+1
    }
    return []int{ans1,ans2}
}